.github/workflows/smart-contract-ci.yml

```diff
@@ -51,8 +51,8 @@ jobs:
         with:
           node-version: ${{ inputs.node-version }}
           cache: "npm"
-          cache-dependency-path: ${{ inputs.working-directory }}/smart_contracts/package-lock.json
-      
+          cache-dependency-path: ./package-lock.json
```

Comment:
Will this work? There is still a package-lock in the sub-directory and I'm not sure if GitHub cache works off of a parent package.json with workspaces.

Reply:
Shouldn't be any package-lock's outside of the root directory and infrastructure(not in workspace). Pretty sure it's supported, worst case scenario we can implement our own cache strategy. Still need to refactor the CI/CD pipeline to be in-line with workspaces. (running install once and recursive running of scripts). 

This is an experimental branch of changes found during discovery and may end up being dropped and reintroduced (depending on backlog pressure). Very much a WIP

Reply:
Sure, makes sense if you are going to house the package-lock at that position. Worth pointing out though that you'll want to check that the output of npm run build for each sub-component works with the cdk code. If you have trouble hit me up.